### PR TITLE
Astro 1946 push button sizes

### PR DESCRIPTION
**Pull request by Cheshire89**
## Brief Description

Add support for push button sizes and icons

## JIRA Link

https://www.figma.com/file/ZtHeSWpISd8hJjN46uY9pm/Astro-UX-Design-System?node-id=4240%3A29475
https://rocketcom.atlassian.net/browse/ASTRO-1946

## Types of changes

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Breaking change

## Checklist

-   [ ] This PR adds or removes a Storybook story.
-   [ ] I have added tests to cover my changes.

### CODE REVIEW

File: src/components/rux-push-button/rux-push-button.scss [hunk 1]
```diff
@@ -44,25 +45,60 @@
...
+            }
         }
 
         &__input:disabled + .rux-push-button__button {
             opacity: var(--disabled-opacity);
             cursor: not-allowed;
         }
     }
+
+    :host([size='small']) {
```
Comment (NicDominguez) [inline src/components/rux-push-button/rux-push-button.scss:97]: Did you mean to keep these empty rules in?

File: src/components/rux-push-button/rux-push-button.scss [hunk 2]
```diff
@@ -44,25 +45,60 @@
...
             border: 1px solid var(--pushbutton-border-color);
+            padding: 0.5rem 1rem;
+
+            &--small {
+                padding: 0.25rem 1rem;
+            }
+
+            &--large {
+                padding: 0.875rem 1rem;
+            }
```
Comment (NicDominguez) [inline src/components/rux-push-button/rux-push-button.scss:60]: According to figma the padding is 3.5px, 7.5px, and 13.5px vertically. I wouldn't normally fret about a 0.5px difference but my main concern is with the rem values. Do you know if design wants a set padding or relative to the root text size? I suppose its fine if the root size is always expected to be 16px.

**Review by NicDominguez — APPROVED** [review]
Nicely done!

File: src/components/rux-push-button/rux-push-button.scss (hunk 2 above)
Comment (Cheshire89) [reply src/components/rux-push-button/rux-push-button.scss:60]: Screen Shot 2021-09-21 at 2 54 56 PM
Screen Shot 2021-09-21 at 2 54 48 PM
Screen Shot 2021-09-21 at 2 54 39 PM
its on point with figma

File: src/components/rux-push-button/rux-push-button.scss (hunk 1 above)
Comment (Cheshire89) [reply src/components/rux-push-button/rux-push-button.scss:97]: No this needs to be trashed

**Pull request merged**